test/malformed_data/test_malformed_events.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import time
+import uuid
+import logging
+import sys
+from common import open_fifo,\
+    write_proto_buf_message, request_start_logging, \
+    check_events_from_splunk, request_stop_logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler(sys.stdout)
+logger.addHandler(handler)
+
+
+@pytest.mark.parametrize("test_input,expected", [
+    ("", 0),  # should not be sent to splunk
+    (" ", 0),  # should not be sent to splunk
+    ("\xF0\xA4\xAD", 1),  # non utf-8 decodable chars should still make it to splunk
+    ("hello", 1),  # normal string should always to sent to splunk
+])
```

Comment:
Should we also add a test for malformed JSON structure with missing quotes or braces? EG: {"test":"value}